aten/src/TH/generic/THTensorRandom.c

```diff
@@ -92,6 +92,17 @@ void THTensor_(uniform)(THTensor *self, THGenerator *_generator, double a, doubl
 
 void THTensor_(normal)(THTensor *self, THGenerator *_generator, double mean, double stdv)
 {
+#if defined(TH_REAL_IS_FLOAT) && defined(USE_AVX2)
+  const int size = THTensor_(numel)(self);
+  if (size >= 16 && THTensor_(isContiguous)(self)) {
+    THTensor_(normal_fill_AVX2)(self->storage->data,
```

Comment:
Does this handle alignment of `data`?

I think we need cpu dispatch of `normal_fill_AVX2`. For example, we distribute binary builds that are compiled with AVX, but need to run on processors without AVX2 support.

Reply:
See THVectorDispatch.c/h for example

Reply:
^^ what Sam says. see vectorDispatchInit in https://github.com/pytorch/pytorch/blob/master/aten/src/TH/generic/THVectorDispatch.c